### PR TITLE
Roe 114 create corporate managing officer page

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-114

### Change description
Created managing officer corporate page with routing, controller and unit testing.

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: src/controllers/managing.officer.corporate.controller.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { Request, Response } from "express";
+import * as config from "../config";
+import { logger } from "../utils/logger";
+
+export const get = (req: Request, res: Response) => {
+  logger.debug(`GET ${config.MANAGING_OFFICER_CORPORATE_PAGE}`);
+  return res.render(config.MANAGING_OFFICER_CORPORATE_PAGE);
```
Comment (SVarallo) [inline src/controllers/managing.officer.corporate.controller.ts:7]: Do you know the page that comes before this right now? Will need to pass in the backLinkUrl if so

File: views/managing-officer-corporate.html [hunk 2]
```diff
@@ -0,0 +1,303 @@
+{% extends "layout.html" %}
+
+{% block pageTitle %}
+  Corporate managing officer – {{ serviceName }} – GOV.UK
```
Comment (markpit) [inline views/managing-officer-corporate.html:4]: need to replace service name with the name of our app (see other templates) and the " – " character is flagged as invalid in the vscode editor  - The character U+2013 "–" could be confused with the character U+002d "-", I would change them to be minus signs.

File: views/managing-officer-corporate.html [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+
+      {{ govukButton({
+        text: "Add another corporate managing officer",
+        href: "",
+        classes: "govuk-button--secondary govuk-!-margin-top-6"
+      }) }}
+
+      {{ govukButton({
+        text: "I do not need to add any more corporate managing officers",
+        href: "/register/check-answers"
```
Comment (markpit) [inline views/managing-officer-corporate.html:295]: button needs a submit attribute - see https://github.com/companieshouse/overseas-entities-web/blob/main/views/beneficial-owner-gov.html#L311

File: test/controllers/managing.officer.corporate.controller.spec.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import { NextFunction, Request, Response } from "express";
+import request from "supertest";
+import app from "../../src/app";
+import { authentication } from "../../src/controllers";
+import { MANAGING_OFFICER_CORPORATE_URL } from "../../src/config";
+
+const mockAuthenticationMiddleware = authentication as jest.Mock;
+mockAuthenticationMiddleware.mockImplementation((req: Request, res: Response, next: NextFunction) => next() );
+
+const MANAGING_OFFICER_CORPORATE_PAGE_TITLE = "Tell us about the corporate managing officer";
```
Comment (markpit) [inline test/controllers/managing.officer.corporate.controller.spec.ts:14]: page heading should go in https://github.com/companieshouse/overseas-entities-web/blob/main/test/__mocks__/text.mock.ts with the others

File: views/managing-officer-corporate.html (hunk 3 above)
Comment (SVarallo) [reply views/managing-officer-corporate.html:295]: Wont the other button need an id as well so the automated tests know the difference between them?

Comment (markpit) [reply views/managing-officer-corporate.html:295]: yes, although i didn't add one on my template (for the first button) not sure what the the id would be as submit would be used for the 'submit' page (second button)

File: views/managing-officer-corporate.html (hunk 2 above)
Comment (markpit) [reply views/managing-officer-corporate.html]: does the {{ serviceName }} get replaced? i had replaced it with text on mine but perhaps I didn't need to, does it get replaced in your docker env?

File: views/managing-officer-corporate.html (hunk 3 above)
Comment (markpit) [reply views/managing-officer-corporate.html]: i think Sam is right, we should add an id attribute for the first button, I'll do the same on my page. I'm going to add an id of 'add' on mine, if you do the same here then they'll be consistent.

**Review by markpit — APPROVED** [review]

**Pull request merged**